.github/workflows/release.yml

```
@@ -0,0 +1,64 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  check-token:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check token variable
+        env:
+          PYPI_TOKEN_VAR: ${{ secrets.pypi_token }}
+        run: |
+          if [ -z "$PYPI_TOKEN_VAR" ]; then
+            echo "Pypi token variable is empty!"
+            exit 1
+          fi
+          echo "Pypi token variable is set!"
+  build:
+    name: Build package
+    needs: check-token
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build a binary wheel and a source tarball
+        run: python -m build
+
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist
+  
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
```

```suggestion
      name: release
```

I've now [setup the PyPI project to use a trusted publisher](https://til.simonwillison.net/pypi/pypi-releases-from-github). The name of the environment is `release`. So please update the name for this particular job.

---

Thank you for the advice.